File: Java/src/test/java/re/nico/async/AsyncTest.java
package re.nico.async;

import static org.junit.jupiter.api.Assertions.assertEquals;
import java.util.concurrent.ExecutionException;
import org.junit.jupiter.api.Test;

final class AsyncTest {

    String expectedResponse = "<html><body>Hi!</body></html>";

    @Test
    void testBasicsFuture() throws InterruptedException, ExecutionException {
        assertEquals(expectedResponse, BasicsFuture.run());
    }

    @Test
    void testAsyncHandlers() throws InterruptedException, ExecutionException {
        assertEquals(expectedResponse, AsyncHandlers.run());
    }

    @Test
    void testCompletableFuture() {
        assertEquals(expectedResponse, CompletableFuture.run());
    }

    @Test
    void testListenableFutureDemo() throws InterruptedException, ExecutionException {
        assertEquals(expectedResponse, ListenableFutureDemo.run());
    }

    @Test
    void testReactiveStreams() throws InterruptedException, ExecutionException, Throwable {
        assertEquals(expectedResponse, ReactiveStreams.run());
    }

}

File: Kotlin/src/test/kotlin/async/AsyncTest.kt
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import channel.*

class AsyncTest {

    var expectedResponse: String = "<html><body>Hi!</body></html>"

    @Test
    fun testGoroutines() {
        var realResult: String = ""
        mainBlocking {
            realResult = goroutines()
        }
        assertEquals(expectedResponse, realResult)
        println("Done!")
    }
}
